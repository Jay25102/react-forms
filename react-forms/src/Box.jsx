import React from 'react';

const Box = function({id, backgroundColor = "black", width = 100, height = 100, handleRemove}) {
    let styles = {
        backgroundColor,
        width: `${width}px`,
        height: `${height}px`,
    }

    const remove = function() {
        handleRemove(id);
    }

    return(
        <div>
            <div
                style={styles}
            ></div>
            <button onClick={remove}>Remove</button>
        </div>
        
    )
}

export default Box;