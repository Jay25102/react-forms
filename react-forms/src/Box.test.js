import Box from "./Box";
import { render } from "@testing-library/react";

describe("Box smoke and snapshot", function() {
    it("renders without crashing", function() {
        render(<Box/>)
    });

    it("matches snapshot", function() {
        const {asFragment} = render(<Box/>);
        expect(asFragment()).toMatchSnapshot();
    });
});