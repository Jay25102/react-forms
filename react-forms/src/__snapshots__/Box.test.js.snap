// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box smoke and snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="background-color: black; width: 100px; height: 100px;"
    />
    <button>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
